applications/ExaquteSandboxApplication/custom_processes/metrics_divergencefree_process.cpp

```diff
@@ -0,0 +1,305 @@
+     * maximal_size: The maximal size to consider on the remeshing
+     * refinement_strategy: The chosen refinement strategy
+     * echo_level: The verbosity
+     *
+     * mean_distribution_strategy
+     * target_refinement_coefficient: Coefficient to make the average element size throughout the domain change by a factor of target_refinement_coefficient after the refinement
+     * refinement_bound: Additional tolerance to bound the density factor (refinement coefficient)
+     * reference_variable_name: The refinement indicator
+     * reference_norm_name: The refinement norm indicator, exploited to estimate some coefficients
+     *
+     * maximum_strategy
+     * target_refinement_coefficient: The range of elements we want to refine w.r.t. the maximum value of reference_variable_name
+     * refinement_coefficient: Coefficient to make the average element size throughout the domain change by a factor of target_refinement_coefficient after the refinement
+     * reference_variable_name: The refinement indicator
+     */
+    Parameters default_parameters = Parameters(R"(
+    {
+        "minimal_size"                        : 0.01,
+        "maximal_size"                        : 10.0,
+        "refinement_strategy"                 : "maximum_strategy",
+        "mean_distribution_strategy":
+        {
+            "target_refinement_coefficient"       : 0.9,
+            "refinement_bound"                    : 2.0,
+            "reference_variable_name"             : "DIVERGENCE_WEIGHTED",
+            "reference_norm_name"                 : "VELOCITY_H1_SEMINORM"
+        },
+        "maximum_strategy":
+        {
+            "target_refinement_coefficient"       : 0.1,
+            "refinement_coefficient"              : 2.0,
+            "reference_variable_name"             : "DIVERGENCE_WEIGHTED"
+        },
+        "echo_level"                          : 0
+    })"
+    );
+
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+    mMinSize = ThisParameters["minimal_size"].GetDouble();
+    mMaxSize = ThisParameters["maximal_size"].GetDouble();
+    mRefinementStrategy = ThisParameters["refinement_strategy"].GetString();
+    mEchoLevel = ThisParameters["echo_level"].GetInt();
+
+    // Mean distribution strategy
+    mMeanStrategyReferenceVariable = ThisParameters["mean_distribution_strategy"]["reference_variable_name"].GetString();
+    mMeanStrategyReferenceNorm = ThisParameters["mean_distribution_strategy"]["reference_norm_name"].GetString();
+    mMeanStrategyTargetRefinementCoefficient = ThisParameters["mean_distribution_strategy"]["target_refinement_coefficient"].GetDouble();
+    mMeanStrategyRefinementBound = ThisParameters["mean_distribution_strategy"]["refinement_bound"].GetDouble();
+    mMeanStrategyDivergenceFreeOverAllDomain = 0;
+
+    // Maximum strategy
+    mMaxStrategyReferenceVariable = ThisParameters["maximum_strategy"]["reference_variable_name"].GetString();
```

Comment:
here youshould convert the input string to an enum, as Vicente commented.

the idea is to have a list of options and to verify that the option you input is admissible.
take a look at how that is done in the constructor of the amgcl_solver

Reply:
I used enum, I think I did it as you intended